src/middlewared/middlewared/plugins/api_key.py

```diff
@@ -58,6 +99,8 @@ async def do_create(self, data):
             data
         )
 
+        await self.load_keys()
```

Comment:
Why load everything again if you can just add/update/remove the key in question?